lib/va_profile/v3/address_validation/address_suggestions_response.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'va_profile/models/v3/validation_address'
+require_relative 'service'
+
+module VAProfile
+  module V3
+    module AddressValidation
+      # Wrapper for response from VA profile address validation API.
+      # Contains address suggestions and validation key used to ignore suggested addresses
+      # and use original address.
+      class AddressSuggestionsResponse
+        def initialize(candidate_res)
+          validation_key = candidate_res['override_validation_key']
+          @response = {
+            addresses: candidate_res['candidate_addresses'].map do |address_suggestion_hash|
+              {
+                address: VAProfile::Models::V3::ValidationAddress.build_from_address_suggestion(
```

Comment:
It seems like we should add some error handling here, otherwise it could cause the entire method to crash if errors are raised.

Maybe something like this:

```
def initialize(candidate_res)
  validation_key = candidate_res['override_validation_key']

  @response = {
    addresses: build_addresses(candidate_res['candidate_addresses']),
    validation_key:
  }
end

private


def build_addresses(candidate_addresses)
  candidate_addresses.map do |address_suggestion_hash|
    begin
      {
        address: build_address(address_suggestion_hash),
        address_meta_data: build_metadata(address_suggestion_hash)
      }
    rescue StandardError => e
      Rails.logger.error("Failed to build address or metadata: #{e.message}")
      # Handle gracefully, perhaps return a fallback value or skip this entry
      { address: {}, address_meta_data: {} }
    end
  end
end

def build_address(address_suggestion_hash)
  VAProfile::Models::V3::ValidationAddress
    .build_from_address_suggestion(address_suggestion_hash)
    .to_h.compact
end

def build_metadata(address_suggestion_hash)
  VAProfile::Models::V3::ValidationAddress
    .build_address_metadata(address_suggestion_hash)
    .to_h
end
```

Reply:
Ya, not sure why they weren't labeled as a private message before.

Reply:
These methods are called outside this model

Reply:
The errors will not be nil. Either the params will be there or they will not. The params directly come from the AddressValidation Api Service

Reply:
The errors are already handled here: https://github.com/department-of-veterans-affairs/vets-api/pull/18895/files#diff-05557e768732eda9c74f84335fc893491706df4e7bf134d11c776463f1317a6dR42